zh, en: update online ddl doc (#734)
pingcap/docs-dm: en/key-features.md

### Restrictions

- DM only supports gh-ost and pt-osc.
- When `online-ddl` is enabled, the checkpoint corresponding to incremental replication should not be in the process of online DDL execution. For example, if an upstream online DDL operation starts at `position-A` and ends at `position-B` of the binlog, the starting point of incremental replication should be earlier than `position-A` or later than `position-B`; otherwise, an error occurs. For details, refer to [FAQ](faq.md#how-to-handle-the-error-returned-by-the-ddl-operation-related-to-the-gh-ost-table-after-online-ddl-scheme-gh-ost-is-set).

### Parameter configuration

- If the upstream MySQL/MariaDB (at the same time) uses the gh-ost or pt-osc tool, set `online-ddl` to `true` in the task configuration file:

```yml
online-ddl: true
```

For more information about online DDL tools, refer to [Online DDL](feature-online-ddl.md).

> **Tip:**
>

> > `online-ddl-scheme` is deprecated, it is recommended to use `online-ddl` instead of `online-ddl-scheme`.

Reviewer: Shall we make it clear from which version we recommend user to use `online-ddl` and below this version, use should still use `online-ddl-scheme`

Reviewer 2: may take a look at "tab" feature

https://docs.pingcap.com/zh/tidb-data-migration/stable/relay-log#%E5%90%AF%E5%8A%A8%E5%81%9C%E6%AD%A2-relay-log

### Restrictions

- DM only supports gh-ost and pt-osc.
- When `online-ddl` is enabled, the checkpoint corresponding to incremental replication should not be in the process of online DDL execution. For example, if an upstream online DDL operation starts at `position-A` and ends at `position-B` of the binlog, the starting point of incremental replication should be earlier than `position-A` or later than `position-B`; otherwise, an error occurs. For details, refer to [FAQ](faq.md#how-to-handle-the-error-returned-by-the-ddl-operation-related-to-the-gh-ost-table-after-online-ddl-scheme-gh-ost-is-set).

### Parameter configuration

- If the upstream MySQL/MariaDB (at the same time) uses the gh-ost or pt-osc tool, set `online-ddl` to `true` in the task configuration file:

```yml
online-ddl: true
```

> - If you use the `online-ddl-scheme`, set it in the task configuration file

Reviewer 3: ```suggestion
- If you use the `online-ddl-scheme`, set it in the task configuration file:
```

> - If you use the `online-ddl-scheme`, set it in the task configuration file:

Reviewer 3: ```suggestion
- If you use the `online-ddl-scheme`, set it as follows in the task configuration file:
```

```yml
online-ddl: false
online-ddl-scheme: "gh-ost"    # Only support gh-ost
```

- Or

```yml
online-ddl: false
online-ddl-scheme: "pt"        # Only support pt
```

For more information about online DDL tools, refer to [Online DDL](feature-online-ddl.md).

> **Tip:**
>

> > It is recommended to use `online-ddl` instead of `online-ddl-scheme`.

Reviewer 3: ```suggestion
> Because `online-ddl-scheme` will be deprecated in the future, it is recommended to use `online-ddl` instead of `online-ddl-scheme`.
```

### Restrictions

- DM only supports gh-ost and pt-osc.
- When `online-ddl` is enabled, the checkpoint corresponding to incremental replication should not be in the process of online DDL execution. For example, if an upstream online DDL operation starts at `position-A` and ends at `position-B` of the binlog, the starting point of incremental replication should be earlier than `position-A` or later than `position-B`; otherwise, an error occurs. For details, refer to [FAQ](faq.md#how-to-handle-the-error-returned-by-the-ddl-operation-related-to-the-gh-ost-table-after-online-ddl-scheme-gh-ost-is-set).

### Parameter configuration

<SimpleTab>
<div label="v2.0.5 and later">
 

> In v2.0.5 and later versions, please use the `online-ddl` configuration item in the `task` configuration file.

Reviewer 3: ```suggestion
In v2.0.5 and later versions, you need to use the `online-ddl` configuration item in the `task` configuration file.
```

- If the upstream MySQL/MariaDB (at the same time) uses the gh-ost or pt-osc tool, set `online-ddl` to `true` in the task configuration file:

```yml
online-ddl: true
```

> **Note:**
>

> > Since v2.0.5, `online-ddl-scheme` has been deprecated, please use `online-ddl` instead of `online-ddl-scheme`. For example, setting `online-ddl: true` will overwrite `online-ddl-scheme`, and setting `online-ddl-scheme: "pt"` or `online-ddl-scheme: "gh-ost"` will also be converted to `online-ddl: true`.

Reviewer 3: ```suggestion
> Since v2.0.5, `online-ddl-scheme` has been deprecated, so you need to use `online-ddl` instead of `online-ddl-scheme`. That means that setting `online-ddl: true` overwrites `online-ddl-scheme`, and setting `online-ddl-scheme: "pt"` or `online-ddl-scheme: "gh-ost"` is converted to `online-ddl: true`.
```

</div>

<div label="earlier than v2.0.5">

> Before v2.0.5 (not including v2.0.5), please use the `online-ddl-scheme` configuration item in the `task` configuration file.

Reviewer 3: ```suggestion
Before v2.0.5 (not including v2.0.5), you need to use the `online-ddl-scheme` configuration item in the `task` configuration file.
```
